doc/source/querying.rst

```diff
@@ -0,0 +1,96 @@
+The :py:class:`freud.locality.NeighborQuery` class defines an abstract interface for neighbor finding that is implemented by its subclasses, namely the :py:class:`freud.locality.LinkCell` and :py:class:`freud.locality.AABBQuery` classes.
+These classes represent specific data structures used to accelerate neighbor finding.
+These two different methods have different performance characteristics, but in most cases :class:`freud.locality.AABBQuery` performs no worse than :class:`freud.locality.LinkCell` and is entirely parameter free, so it is the default method of choice used internally in **freud**'s ``PairCompute classes``.
+
+In general, these data structures operate by constructing them using one set of points, after which they can be queried to efficiently find the neighbors of arbitrary other points using :py:meth:`freud.locality.NeighborQuery.query`.
+
+
+Query Arguments
+===============
+
+The table below describes the set of valid query arguments.
+
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| Query Argument | Definition                                                            | Data type | Legal Values              | Valid for                                                           |
++================+=======================================================================+===========+===========================+=====================================================================+
+| mode           | The type of query to perform (distance cutoff or number of neighbors) | str       | 'none', 'ball', 'nearest' | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| r_max          | Maximum distance to find neighbors                                    | float     | r_max > 0                 | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| r_min          | Minimum distance to find neighbors                                    | float     | 0 <= r_min < r_max        | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| num_neighbors  | Number of Neighbors                                                   | int       | num_neighbors >= 0        | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
```

Comment:
Does 0 actually work? Have we tested that edge case?

Reply:
Not sure... I mean to just have >0, I'll change.